### DISCUSSION TITLE
Is there any issue if `autoModeWatcher` for both ssr and non-ssr are added in root `+layout.svelte` ?

### DISCUSSION THREAD

**Discussion by Haiz14** [skeletonlabs/skeleton#2801] [Q&A]
On the website i am working at, some pages are server-rendered, some are client-rendered.

If i add `autoModeWatcher` in root `+layout.svelte` for both ssr and non ssr, will it cause any issue?

```svelte
<script>
  import { onMount } from 'svelte';
  import { autoModeWatcher } from '@skeletonlabs/skeleton';
  onMount(() => {						 
    autoModeWatcher();
  })
</script>

<svelte:head>{@html '<script>(' + autoModeWatcher.toString() + ')();</script>'}</svelte:head>
```

**Comment by Haiz14** [comment]
I did use both, and everything seems to work fine on both csr and SSR pages.

**Comment by Hugos68** [comment]
Hey @Haiz14, since the `autoModeWatcher` runs in the `head` it will only work on the client. However, this is intended, as mode toggling always happens on the client, not the server.

So the the `autoModeWatcher` is both SSR and Client safe.

I also happen to notice you did:
```js
  onMount(() => {						 
    autoModeWatcher();
  })
```
This can be removed, as the `autoModeWatcher` already runs inside the `script` tag within the `head`.

If you *do* encounter any issues, please let us know!

**Comment by Haiz14** [comment]
Selected light or dark mode is breaking on CSR pages. I think I would have to add onMount in all CSR pages.

- visit a non CSR pages change light mode
- then visit CSR pages, change dark mode
- on that same page, change mode back to light
- reload this CSR page, the selected mode is light, but displayed mode is dark.

**Reply by Hugos68** [reply]
Well it won't work on a non CSR page, you're disabling javascript, and you need javascript for the mode watcher. You were talking about SSR, disabling CSR is not an option if you want to use it.

**Comment by Haiz14** [comment]
Example of broken mode
![Screenshot_2024-08-12-14-36-07-21_40deb401b9ffe8e1df2f1cc5ba480b12](https://github.com/user-attachments/assets/013eeb5c-2c81-4991-a439-48075f1aebb9)
